RevenueCatUI/CustomerCenter/ViewModels/PromotionalOfferViewModel.swift

```diff
@@ -40,12 +40,12 @@ class PromotionalOfferViewModel: ObservableObject {
 
     init(
         promotionalOfferData: PromotionalOfferData?,
-        purchasesProvider: CustomerCenterPurchasesType = CustomerCenterPurchases(),
+        purchasesProvider: CustomerCenterPurchasesType,
```

Comment:
I kind of dislike to give up this convenience, but I guess the dependency injection pattern is a broader discussion, out of scope for sure 😭

Reply:
Well, we can totally keep the default parameters for convenience. But at the same time, we need to make sure that we're passing the dependency when we have it. 
It seems to me that we only want the convenience for previews, and perhaps that shouldn't be the leading factor in this case 🤔  WDYT?